aiida/cmdline/commands/cmd_verdi.py

```diff
@@ -14,36 +14,27 @@
 
 import click
 
-from aiida.cmdline.params import options
+from aiida.cmdline.params import options, types
 from aiida.common.extendeddicts import AttributeDict
-from aiida.common import exceptions
 
 
 @click.group(context_settings={'help_option_names': ['-h', '--help']})
-@options.PROFILE()
+@options.PROFILE(type=types.ProfileParamType(load_profile=True))
 @click.version_option(None, '-v', '--version', message="AiiDA version %(version)s")
 @click.pass_context
 def verdi(ctx, profile):
     """The command line interface of AiiDA."""
-    from aiida.manage.configuration import get_config, load_profile
+    from aiida.manage.configuration import get_config, get_profile
 
     if ctx.obj is None:
         ctx.obj = AttributeDict()
 
     config = get_config(create=True)
+    profile = get_profile()
```

Comment:
why not use the profile that is returned by the option?

Reply:
Because the option will return `None` if no specific profile was specified. However, other commands depend on `ctx.obj.profile` to contain the profile that is to be used.

Reply:
I see - is this the behavior we want, though?
If somebody else wants to reuse the `--profile` option, I guess it would be natural to want to get the default profile, if non is specified.
I.e. can't we make the default behavior to load the default profile?

Reply:
I thought about this and think not (but not 100% sure), because it won't be possible to know whether the `-p` flag was explicitly set or the default was used, which is what we set in `ctx.obj.profile_option_used`. This boolean is now used for example in the `get_default_profile` function which is used by `verdi profile show` as callback for the profile argument. Let's say default profile is `django`. If I call `verdi profile show` it will show that of `django`. But if I call `verdi -p sqla profile show` it will still show details of django, if it weren't for the `ctx.obj.profile_option_used`.

This is currently used only in this callback function. I will try now if I can get rid of it and have the `PROFILE` option to use a default